### ISSUE TITLE
non placeholder query param not handled correctly when its name is same as a placeholder

### ISSUE THREAD

**Issue by bNobo** [ThreeMammals/Ocelot#1450]
## Expected Behavior / New Feature
Given this downstream route : `/api/units/{subscriptionId}/{unitId}/updates`
When the request uri is `http://localhost:5000/api/subscriptions/1/updates?unitId=2&productId=2&subscriptionId=1`
Then the dowstream request URI should be `https://localhost:5000/api/units/1/2/updates?productId=2&subscriptionId=1`

## Actual Behavior / Motivation for New Feature
Actually the downstream request is `https://localhost:5000/api/units/1/2/updates?productId=2subscriptionId=1`

The "&" before "subscriptionId" is missing

## Steps to Reproduce the Problem

You can easily reproduce this problem with a slightly modified version of an existing unit test:

  1. Checkout this branch from my fork https://github.com/bNobo/Ocelot/tree/arg-with-placeholder-name
  1. Execute `should_replace_query_string_but_leave_non_placeholder_queries` unit test
 
Then you should see this error  message:

```
Message: 
    Shouldly.ShouldAssertException : _httpContext.Items.DownstreamRequest().ToHttpRequestMessage().RequestUri.OriginalString
        should be
    "https://localhost:5000/api/units/1/2/updates?productId=2&subscriptionId=1"
        but was
    "https://localhost:5000/api/units/1/2/updates?productId=2subscriptionId=1"
        difference
```
![image](https://user-images.githubusercontent.com/11414291/110317494-8092cf80-800c-11eb-8a89-7143097e981e.png)

## Specifications

  - Version: 17.0.0
  - Platform: Windows
  - Subsystem:

The issue also occurs on version 16.0.2

**Comment by bNobo** [comment]
Maybe I can solve this problem by myself ? Do you have advices for me on the easiest way to solve it ?

**Comment by jlukawska** [comment]
It seems to be a similar (or the same) problem to #1288 and probably PR #1335 will solve it. Unfortunately, it's been waiting a long time...

**Comment by bNobo** [comment]
I've tested the solution proposed in PR #1335 on my branch and it does not solve my problem. It seems that #1288 is a similar but not exactly the same problem.

**Comment by bNobo** [comment]
I don't see a way to fix it : if I modify the code in order to not remove the ampersand before the "regex replace", then the query string `subscriptionId=1` will be removed from URL because it is the purpose of `RemoveQueryStringParametersThatHaveBeenUsedInTemplate` to do that. 

In my case I need to keep the query string in URL despite the fact it is already part of the path. It is a requirement for a legacy application...

This case seems to be unmanageable! 

But in the short run I have a workaround : I've modified my routes to rename every placeholder with a "leet speech" string as name. For instance if I had a placeholder like {version} I replaced it with {V3R510N}. Thus placeholders stay "almost" easy to read and it is unlikely that an additionnal parameter would have the same name...

**Comment by jlukawska** [comment]
Ocelot has a collection of placeholder pairs (name, value) and it tries to remove placeholders from query string if it finds them. That's why the query param name and value cannot be the same as a placeholder name and value if you want to keep it in the query string. 
If you want to keep the whole query string you can try to comment out RemoveQueryStringParametersThatHaveBeenUsedInTemplate method, but I'm not sure if there are any side effects.

By the way, the result of your example should remove not only "&", but also "subscriptionId=1" (according to the current implementation). I've just fixed it in #1335  (unfortunately it won't help you, but it's a bug).

**Comment by bNobo** [comment]
Effectively I'm afraid they'd have some side effects to comment out RemoveQueryStringParametersThatHaveBeenUsedInTemplate, that's why I prefer to apply a workaround playing on routes until the legacy application is fixed.

So the root of this issue is not Ocelot but the legacy application. 

I'm glad this issue helped to solve a bug, I can close it now :)
